File: src/#938/solution.go
package solution

type TreeNode struct {
	Val         int
	Left, Right *TreeNode
}

func rangeSumBST(root *TreeNode, L int, R int) int {
	if root == nil {
		return 0
	}
	v := 0
	if root.Val >= L && root.Val <= R {
		v = root.Val
	}
	return v + rangeSumBST(root.Left, L, R) + rangeSumBST(root.Right, L, R)
}

File: src/#938/solution.js
function TreeNode(val, left, right) {
  this.val = val === undefined ? 0 : val
  this.left = left === undefined ? null : left
  this.right = right === undefined ? null : right
}

var rangeSumBST = function (root, L, R) {
  if (root === null) {
    return 0
  }
  let v = root.val >= L && root.val <= R ? root.val : 0
  return v + rangeSumBST(root.left, L, R) + rangeSumBST(root.right, L, R)
}
